From e5be3047681bcee43091276439642dd52321328d Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Wed, 2 Apr 2014 15:18:00 -0500
Subject: [PATCH 229/248] FogBugz #195004: Move board specific components to
 board dts files

Move board components from socfpga_cyclone5.dtsi and socfpga_arria5.dtsi
into the boards specific files.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/boot/dts/socfpga_arria5.dtsi        |  47 -----------
 arch/arm/boot/dts/socfpga_arria5_socdk.dts   |  43 ++++++++++
 arch/arm/boot/dts/socfpga_cyclone5.dtsi      | 115 ---------------------------
 arch/arm/boot/dts/socfpga_cyclone5_socdk.dts |  90 ++++++++++++++++++++-
 4 files changed, 132 insertions(+), 163 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga_arria5.dtsi b/arch/arm/boot/dts/socfpga_arria5.dtsi
index ac46fed..8f6d98c 100644
--- a/arch/arm/boot/dts/socfpga_arria5.dtsi
+++ b/arch/arm/boot/dts/socfpga_arria5.dtsi
@@ -19,55 +19,8 @@
 
 / {
 	soc {
-		dwmmc0@ff704000 {
-			num-slots = <1>;
-			supports-highspeed;
-			broken-cd;
-			altr,dw-mshc-ciu-div = <3>;
-			altr,dw-mshc-sdr-timing = <0 3>;
-
-			slot@0 {
-				reg = <0>;
-				bus-width = <4>;
-			};
-		};
-
-		gpio@ff708000 {
-			status = "okay";
-		};
-
-		gpio@ff709000 {
-			status = "okay";
-		};
-
-		gpio@ff70a000 {
-			status = "okay";
-		};
-
-		i2c0: i2c@ffc04000 {
-			speed-mode = <0>;
-			status = "okay";
-		};
-
-		qspi: spi@ff705000 {
-			status = "okay";
-		};
-
-		serial0@ffc02000 {
-			status = "okay";
-		};
-
 		sysmgr@ffd08000 {
 			cpu1-start-addr = <0xffd080c4>;
 		};
-
-		usb1: usb@ffb40000 {
-			status = "okay";
-		};
-
-		watchdog0: wd@ffd02000 {
-			status = "okay";
-		};
-
 	};
 };
diff --git a/arch/arm/boot/dts/socfpga_arria5_socdk.dts b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
index bec3794..b4c4e3e 100644
--- a/arch/arm/boot/dts/socfpga_arria5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
@@ -59,6 +59,20 @@
 			gpios = <&gpio0 18 1>;
 		};
 	};
+
+	soc {
+		gpio@ff708000 {
+			status = "okay";
+		};
+
+		gpio@ff709000 {
+			status = "okay";
+		};
+
+		gpio@ff70a000 {
+			status = "okay";
+		};
+	};
 };
 
 &gmac1 {
@@ -78,6 +92,9 @@
 };
 
 &i2c0 {
+	status = "okay";
+	speed-mode = <0>;
+
 	lcd: lcd@28 {
 		compatible = "newhaven,nhd-0216k3z-nsw-bbw";
 		reg = <0x28>;
@@ -98,11 +115,25 @@
 	};
 };
 
+&mmc {
+	num-slots = <1>;
+	supports-highspeed;
+	broken-cd;
+	altr,dw-mshc-ciu-div = <3>;
+	altr,dw-mshc-sdr-timing = <0 3>;
+
+	slot@0 {
+		reg = <0>;
+		bus-width = <4>;
+	};
+};
+
 &osc1 {
 	clock-frequency = <25000000>;
 };
 
 &qspi {
+	status = "okay";
 	flash0: n25q512a@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -131,3 +162,15 @@
 		};
 	};
 };
+
+&uart0 {
+	status = "okay";
+};
+
+&usb1 {
+	status = "okay";
+};
+
+&watchdog0 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dtsi b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
index 01985f1..25c4b97 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dtsi
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
@@ -18,124 +18,9 @@
 /include/ "socfpga.dtsi"
 
 / {
-	model = "Altera SOCFPGA Cyclone V";
-	compatible = "altr,socfpga-cyclone5", "altr,socfpga";
-
-	chosen {
-		bootargs = "console=ttyS0,115200 root=/dev/mmcblk0p2 rw rootwait";
-	};
-
-	memory {
-		name = "memory";
-		device_type = "memory";
-		reg = <0x0 0x40000000>; /* 1 GB */
-	};
-
-	aliases {
-		/* this allow the ethaddr uboot environmnet variable contents
-		 * to be added to the gmac1 device tree blob.
-		 */
-		ethernet0 = &gmac1;
-	};
-
 	soc {
-		clkmgr@ffd04000 {
-			clocks {
-				osc1 {
-					clock-frequency = <25000000>;
-				};
-			};
-		};
-
-		dcan0: d_can@ffc00000 {
-			status = "okay";
-		};
-
-		dwmmc0@ff704000 {
-			num-slots = <1>;
-			supports-highspeed;
-			broken-cd;
-			altr,dw-mshc-ciu-div = <3>;
-			altr,dw-mshc-sdr-timing = <0 3>;
-
-			slot@0 {
-				reg = <0>;
-				bus-width = <4>;
-			};
-		};
-
-		gpio@ff708000 {
-			status = "okay";
-		};
-
-		gpio@ff709000 {
-			status = "okay";
-		};
-
-		gpio@ff70a000 {
-			status = "okay";
-		};
-
-		i2c0: i2c@ffc04000 {
-			speed-mode = <0>;
-			status = "okay";
-		};
-
-		qspi: spi@ff705000 {
-			status = "okay";
-		};
-
-		serial0@ffc02000 {
-			status = "okay";
-		};
-
 		sysmgr@ffd08000 {
 			cpu1-start-addr = <0xffd080c4>;
 		};
-
-		usb1: usb@ffb40000 {
-			status = "okay";
-		};
-
-		watchdog0: wd@ffd02000 {
-			status = "okay";
-		};
 	};
 };
-
-&i2c0 {
-	lcd: lcd@28 {
-		compatible = "newhaven,nhd-0216k3z-nsw-bbw";
-		reg = <0x28>;
-		height = <2>;
-		width = <16>;
-		brightness = <8>;
-	};
-
-	eeprom@51 {
-		compatible = "atmel,24c32";
-		reg = <0x51>;
-		pagesize = <32>;
-	};
-
-	rtc@68 {
-		compatible = "dallas,ds1339";
-		reg = <0x68>;
-	};
-};
-
-&gmac1 {
-	phy-mode = "rgmii";
-	snps,phy-addr = <0xffffffff>; /* probe for phy addr */
-	rxd0-skew-ps = <0>;
-	rxd0-skew-ps = <0>;
-	rxd1-skew-ps = <0>;
-	rxd2-skew-ps = <0>;
-	rxd3-skew-ps = <0>;
-	txen-skew-ps = <0>;
-	txc-skew-ps = <2600>;
-	rxdv-skew-ps = <0>;
-	rxc-skew-ps = <2000>;
-	max-frame-size = <3800>;
-	status = "okay";
-};
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
index 891536b..c5b4ac0 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
@@ -21,7 +21,7 @@
 	compatible = "altr,socfpga-cyclone5", "altr,socfpga";
 
 	chosen {
-		bootargs = "console=ttyS0,115200";
+		bootargs = "console=ttyS0,115200 root=/dev/mmcblk0p2 rw rootwait";
 	};
 
 	memory {
@@ -59,9 +59,85 @@
 			gpios = <&gpio1 12 1>;
 		};
 	};
+
+	soc {
+		gpio@ff708000 {
+			status = "okay";
+		};
+
+		gpio@ff709000 {
+			status = "okay";
+		};
+
+		gpio@ff70a000 {
+			status = "okay";
+		};
+	};
+};
+
+&dcan0 {
+	status = "okay";
+};
+
+&gmac1 {
+	phy-mode = "rgmii";
+	snps,phy-addr = <0xffffffff>; /* probe for phy addr */
+	rxd0-skew-ps = <0>;
+	rxd0-skew-ps = <0>;
+	rxd1-skew-ps = <0>;
+	rxd2-skew-ps = <0>;
+	rxd3-skew-ps = <0>;
+	txen-skew-ps = <0>;
+	txc-skew-ps = <2600>;
+	rxdv-skew-ps = <0>;
+	rxc-skew-ps = <2000>;
+	max-frame-size = <3800>;
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	speed-mode = <0>;
+
+	lcd: lcd@28 {
+		compatible = "newhaven,nhd-0216k3z-nsw-bbw";
+		reg = <0x28>;
+		height = <2>;
+		width = <16>;
+		brightness = <8>;
+	};
+
+	eeprom@51 {
+		compatible = "atmel,24c32";
+		reg = <0x51>;
+		pagesize = <32>;
+	};
+
+	rtc@68 {
+		compatible = "dallas,ds1339";
+		reg = <0x68>;
+	};
+};
+
+&mmc {
+	num-slots = <1>;
+	supports-highspeed;
+	broken-cd;
+	altr,dw-mshc-ciu-div = <3>;
+	altr,dw-mshc-sdr-timing = <0 3>;
+
+	slot@0 {
+	reg = <0>;
+	bus-width = <4>;
+	};
+};
+
+&osc1 {
+	clock-frequency = <25000000>;
 };
 
 &qspi {
+	status = "okay";
 	flash0: n25q00@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -89,3 +165,15 @@
 		};
 	};
 };
+
+&uart0 {
+	status = "okay";
+};
+
+&usb1 {
+	status = "okay";
+};
+
+&watchdog0 {
+	status = "okay";
+};
-- 
1.9.1

